nodestream_plugin_neptune/neptune_connection.py

```diff
@@ -0,0 +1,115 @@
+import json
+from abc import ABC, abstractmethod
+from logging import getLogger
+
+from aiobotocore.session import get_session
+
+
+class NeptuneConnection(ABC):
+    async def execute(self, query_stmt: str, parameters):
+        response = None
+        async with self._create_boto_client() as client:
+            try:
+                response = await self._execute_query(client,
+                    query_stmt=query_stmt,
+                    parameters=parameters,
+                )
+
+                code = response["ResponseMetadata"]["HTTPStatusCode"]
+                if code != 200:
+                    self.logger.error(
```

Comment:
nit:
to ensure `logger` is always set, is it possible to add something like this?
```
@property
    def logger(self):
        raise NotImplementedError
```

Reply:
Would this not simply shift a failure from an `AttributeError` to a `NotImplementedError`?

Reply:
I think we can simply hoist `logger` to the base class. 

```python
@property
def logger(self):
     return getLogger(self.__class__.__name__)
```